### PR TITLE
release-cli: use Github M1 runner instead of our "m1"

### PR DESCRIPTION
**Pull request by sluongng**
The github runner has Bazelisk preinstalled, so we could get rid of our
custom Bazelisk installation step as well.

### CODE REVIEW

**Comment by sluongng** [comment]
Tested with https://github.com/buildbuddy-io/buildbuddy/actions/runs/7726152613/

**Review by bduffany — APPROVED** [review]
LGTM

sanity check: do the CLI release artifacts from your test run have the correct architecture? (I couldn't find it on the releases page)

File: .github/workflows/release-cli.yaml [hunk 1]
```diff
@@ -115,17 +115,10 @@ jobs:
...
         include:
           - os: ubuntu-20.04
-            shell: bash
-          - os: macos-11
-            shell: bash
-          - os: m1
-            # TODO: Can we remove this if we upgrade our m1 runner to native arm64?
-            shell: /usr/bin/arch -arch arm64e bash --noprofile --norc -eo pipefail {0}
+          - os: macos-13
+          - os: macos-13-xlarge
```
Comment (bduffany) [inline .github/workflows/release-cli.yaml:119]: maybe add a comment here that one of these is x86_64 and one is arm64

**Comment by sluongng** [comment]
Yup it does have the right architecture https://github.com/buildbuddy-io/buildbuddy/actions/runs/7726152613/job/21061866087#step:1:8

We are using the same runner for the weekly M1 release workflows.

Let me add a comment.

File: .github/workflows/release-cli.yaml (hunk 1 above)
Comment (sluongng) [reply .github/workflows/release-cli.yaml]: done

**Pull request merged**